test/extended/two_node/common.go

```diff
@@ -42,3 +44,184 @@ func isClusterOperatorDegraded(operator *v1.ClusterOperator) bool {
+		return "", fmt.Errorf("failed to discover constraint ID using jq on node %s: %v", nodeName, err)
+	}
+
+	constraintId = strings.TrimSpace(constraintId)
+	if constraintId == "" || constraintId == "null" {
+		framework.Logf("Constraint ID not found for resource %s avoiding node %s", resourceName, targetNode)
+		return "", fmt.Errorf("constraint ID not found for resource %s avoiding node %s", resourceName, targetNode)
+	}
+
+	framework.Logf("Found constraint ID: %s for resource %s avoiding node %s", constraintId, resourceName, targetNode)
+	return constraintId, nil
+}
+
+// isResourceStopped checks if a Pacemaker resource is stopped on a specific node
+func isResourceStopped(oc *util.CLI, nodeName string, resourceName string) (bool, error) {
+	framework.Logf("Checking if resource %s is stopped on node %s", resourceName, nodeName)
+
+	cmd := []string{
+		"pcs", "status", "resources", resourceName, fmt.Sprintf("node=%s", nodeName),
+	}
+
+	output, err := util.DebugNodeRetryWithOptionsAndChroot(oc, nodeName, "openshift-etcd", cmd...)
+	if err != nil {
+		return false, fmt.Errorf("failed to get pcs status resources on node %s: %v", nodeName, err)
+	}
+
+	framework.Logf("PCS status resources output:\n%s", output)
+
+	// Resource is stopped if output contains "no active resources"
+	if strings.Contains(strings.ToLower(output), "no active resources") {
+		framework.Logf("Resource %s is stopped on node %s (no active resources)", resourceName, nodeName)
+		return true, nil
+	}
+
+	// Parse each line to check the specific node status
+	lines := strings.Split(output, "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+
+		// Look for lines containing the resource name
+		if strings.Contains(line, resourceName) {
+			// Check if this line shows the target node as Started
+			if strings.Contains(line, "Started") && strings.Contains(line, nodeName) {
+				framework.Logf("Resource %s is running on node %s (Started)", resourceName, nodeName)
+				return false, nil
+			}
+
+			// Check if this line shows the target node as Stopped
+			if strings.Contains(line, "Stopped") && strings.Contains(line, nodeName) {
+				framework.Logf("Resource %s is stopped on node %s (Stopped)", resourceName, nodeName)
+				return true, nil
+			}
+		}
+	}
+
+	// Unknown state - log the output for debugging
+	framework.Logf("Unknown resource state for %s on node %s. Output: %s", resourceName, nodeName, output)
+	return false, fmt.Errorf("could not determine resource %s state on node %s", resourceName, nodeName)
+}
+
+// isKubeletResourceStopped is a convenience function to check if kubelet-clone resource is stopped
+func isKubeletResourceStopped(oc *util.CLI, nodeName string) (bool, error) {
+	return isResourceStopped(oc, nodeName, "kubelet-clone")
+}
+
+// addConstraintAndGetId adds a constraint and returns the constraint ID for later removal
+func addConstraintAndGetId(oc *util.CLI, nodeName string, resourceName string, targetNode string) (string, error) {
```

Comment:
It seems this is not used (actually some other functions seem defined and not used)